service/geopmdpy_test/TestActiveSessions.py

```diff
@@ -0,0 +1,648 @@
+            # write a json file as a dictionary
+            else:
+                json.dump(contents, file)
+
+    def test_default_creation(self):
+        """Default creation of an ActiveSessions object
+
+        Test creates an ActiveSessions object when the geopm-service
+        directory is not present.
+
+        """
+        sess_path = f'{self._TEMP_DIR.name}/geopm-service'
+        act_sess = ActiveSessions(sess_path)
+        self.check_dir_perms(sess_path)
+
+    def test_creation_link_not_dir(self):
+        """The path specified is a link not a directory.
+
+        Test creates an ActiveSessions object when the geopm-service
+        provided path is not a directory, but a link. It asserts that
+        a warning message is printed to standard error and that a
+        new directory is created while the link is renamed.
+
+        """
+        sess_path = f'{self._TEMP_DIR.name}/geopm-service-link'
+        os.symlink('/root', sess_path)
+        with mock.patch('os.path.islink', return_value=True), \
+             mock.patch('os.path.isdir', return_value=False), \
+             mock.patch('uuid.uuid4', return_value='uuid4'), \
+             mock.patch('sys.stderr.write', return_value=None) as mock_err:
+            act_sess = ActiveSessions(sess_path)
+            renamed_path = f'{sess_path}-uuid4-INVALID'
+            calls = [
+                mock.call(f'Warning: <geopm-service> {sess_path} is a symbolic link, the link will be renamed to {renamed_path}'),
+                mock.call(f'Warning: <geopm-service> the symbolic link points to /root')
+            ]
+            for cc, mc in zip(calls, mock_err.call_args_list):
+                self.assertEqual(cc, mc)
+        self.assertTrue(os.path.exists(renamed_path))
+        self.check_dir_perms(sess_path)
+
+    def test_creation_file_not_dir(self):
+        """The path specified is a file not a directory.
+
+        Test creates an ActiveSessions object when the geopm-service
+        provided path is not a directory, but a file. It asserts that
+        a warning message is printed to standard error and that a
+        new directory is created while the file is renamed.
+
+        """
+        sess_path = f'{self._TEMP_DIR.name}/geopm-service.txt'
+        filename = Path(sess_path)
+        filename.touch(exist_ok=True)
+        with mock.patch('os.path.islink', return_value=False), \
+             mock.patch('os.path.isdir', return_value=False), \
+             mock.patch('uuid.uuid4', return_value='uuid4'), \
+             mock.patch('sys.stderr.write', return_value=None) as mock_err:
+            act_sess = ActiveSessions(sess_path)
+            renamed_path = f'{sess_path}-uuid4-INVALID'
+            calls = [
+                mock.call(f'Warning: <geopm-service> {sess_path} is not a directory, it will be renamed to {renamed_path}')
+            ]
+            for cc, mc in zip(calls, mock_err.call_args_list):
+                self.assertEqual(cc, mc)
```

Comment:
```suggestion
            msg = f'Warning: <geopm-service> {sess_path} is not a directory, it will be renamed to {renamed_path}'
            mock_err.assert_called_with(msg)
```